api/src/groups/api.py

```diff
@@ -5,527 +5,507 @@
+@groups.route('/meetings/<int:meeting_id>', methods=['PATCH'])
+@jwt_required
+def update_meeting(meeting_id):
+    meeting_schema = MeetingSchema()
+    try:
+        valid_attributes = meeting_schema.load(request.json, partial=True)
+    except ValidationError as err:
+        return jsonify(err.messages), 422
 
+    return modify_entity(Meeting, meeting_schema, meeting_id, valid_attributes)
 
-@groups.route('/meetings/deactivate/<meeting_id>', methods=['PUT'])
+@groups.route('/meetings/<int:meeting_id>', methods=['DELETE'])
 @jwt_required
-def deactivate_meeting(meeting_id):
+def delete_meeting(meeting_id):
     meeting = db.session.query(Meeting).filter_by(id=meeting_id).first()
 
     if meeting is None:
-        return jsonify(msg="Meeting not found"), 404
+        return jsonify(f"Meeting with id #{meeting_id} does not exist"), 404
 
-    setattr(meeting, 'active', False)
+    db.session.delete(meeting)
     db.session.commit()
-    return jsonify(meeting_schema.dump(meeting))
 
+    # no content should be in the response with 204 status code
+    return "Deleted successfully", 204
 
 # ---- Member
 
 member_schema = MemberSchema()
 
 
-def generate_member(group_id, person_id, joined, active):
-    member = {}
-    member['group_id'] = group_id
-    member['person_id'] = person_id
-    member['joined'] = joined
-    member['active'] = active
-    member = Member(**member_schema.load(member))
-    return member
-
-
-@groups.route('/members', methods=['POST'])
+@groups.route('/groups/<int:group_id>/members', methods=['POST'])
 @jwt_required
-def create_member():
-    request.json['active'] = True
+def create_member(group_id):
+    member_schema = MemberSchema(exclude=['group_id'])
     try:
-        valid_member = member_schema.load(request.json)
+        valid_member = member_schema.load(request.json, partial=['joined', 'active']) # make joined and active optional fields
     except ValidationError as err:
         return jsonify(err.messages), 422
 
-    if db.session.query(Member).filter_by( \
-            group_id=valid_member["group_id"],
-            person_id=valid_member["person_id"]
-    ).count() != 0:
-        return 'member already exists', 409
+    person_id = valid_member['person_id']
+    if db.session.query(Member).filter_by(person_id=person_id, group_id=group_id).first():
+        # if the same member already exists
+        return jsonify(f"Member with group_id #{group_id} and person_id #{person_id} already exists"), 409
+
+    if 'joined' not in valid_member:
+        valid_member['joined'] = datetime.date.today()
+
+    if 'active' not in valid_member:
+        valid_member['active'] = True
 
-    new_member = Member(**valid_member)
+    new_member = Member(group_id=group_id, **valid_member)
     db.session.add(new_member)
-    db.session.commit()
+    try:
+        db.session.commit()
+    except IntegrityError:
+        return jsonify('Payload contains an invalid foreign key'), 404
```

Comment:
This message appears numerous times in this one file. As mentioned, if we are gong to report such errors, they should have some meaning to the end user. For example, we could figure out which FK was invalid and try to map that back into a message that would be helpful.

Another problem with these error messages is that they are not currently internationalized. Please create an issue to review these messages and come up with I18N information for them. In particular, we'll have to figure out how to include parameters in the messages rather than just fixed text in the localization.

Reply:
As we discussed in the stand-up, please open a new issue for error handling and reporting (if you've not done so already).

Reply:
see #530